libnetdata/dictionary/dictionary.c

```diff
@@ -982,6 +982,22 @@ static size_t dictionary_unittest_set_clone(DICTIONARY *dict, char **names, char
     return errors;
 }
 
+static size_t dictionary_unittest_set_null(DICTIONARY *dict, char **names, char **values, size_t entries) {
+    (void)values;
+    size_t errors = 0;
+    size_t i = 0;
+    for(; i < entries ;i++) {
+        char *val = (char *)dictionary_set(dict, names[i], NULL, 0);
+        if(val == NULL) { fprintf(stderr, ">>> %s() returns a non NULL value\n", __FUNCTION__); errors++; }
```

Comment:
This should be null so its probable `(val != NULL)`

Reply:
nice catch!

After spending an hour to understand how it is possible not to have any errors, I understood that the function `dictionary_set()` was declared as `NEVERNULL`, so the compiler just omitted that statement!